opentelemetry-config/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+//! # Library for declarative configuration of OpenTelemetry.
+//!
+//! This library provides a way to configure OpenTelemetry SDK components
+//! using a declarative approach. It allows users to define configurations
+//! for metrics, traces, and exporters in a structured manner.
+
+use std::{
+    collections::HashMap,
+    error::{self, Error},
+    fmt::{self, Display},
+};
+
+use opentelemetry_sdk::{
+    logs::SdkLoggerProvider,
+    metrics::{MeterProviderBuilder, SdkMeterProvider},
+    trace::SdkTracerProvider,
+};
+use serde_yaml::Value;
+
+pub mod model;
+pub mod providers;
+
+/// Registry for different configuration providers.
+pub struct ConfigurationProvidersRegistry {
+    metrics: MetricsProvidersRegistry,
```

Comment:
1. Let's use the name `MeterProvider` to be consistent with the term that the SDK uses.
2. Use the singular form of the word since there is only one provider that we configure.

```suggestion
    metrics: MeterProviderRegistry,
```